pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -233,3 +234,72 @@ def brick_sort_parallel(array, num_threads, **kwargs):
 
     if _check_type(array, DynamicArray):
         array._modify(force=True)
+
+def _matrix_multiply_helper(m1, m2, row, col):
+    s = 0
+    for i in range(len(m1)):
+        s += m1[row][i] * m2[i][col]
+    return s
+
+def matrix_multiply_parallel(matrix_1, matrix_2, num_threads=None):
+    """
+    Implementes concurrent Matrix multiplication
+
+    Parameters
+    ==========
+
+    matrix_1: list of list / tuple of tuple / OneDimensionalArray of OneDimensionalArray
+        Left matrix
+
+    matrix_1: list of list / tuple of tuple / OneDimensionalArray of OneDimensionalArray
+        Right matrix
+
+    num_threads: int
+        The maximum number of threads
+        to be used for multiplication.
+        If not specified then assumed to be equal to number of rows.
+
+    Raises
+    ======
+
+    IndexError
+        When the columns in matrix_1 are not equal to the rows in matrix_2
+
+    Returns
+    =======
+
+    Matrix: Resultant matrix of matrix_1 * matrix_2
+
+    Examples
+    ========
+
+    >>> from pydatastructs import matrix_multiply_parallel
+    >>> I = [[1, 1, 0], [0, 1, 0], [0, 0, 1]]
+    >>> J = [[2, 1, 2], [1, 2, 1], [2, 2, 2]]
+    >>> print(matrix_multiply_parallel(I, J))
+    [[3, 3, 3], [1, 2, 1], [2, 2, 2]]
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Matrix_multiplication_algorithm#Parallel_and_distributed_algorithms
+    """
+    row_matrix_1, col_matrix_1 = len(matrix_1), len(matrix_1[0])
+    row_matrix_2, col_matrix_2 = len(matrix_2), len(matrix_2[0])
+
+    if col_matrix_1 != row_matrix_2:
+        raise IndexError("Matrices cannot be multiplied")
+
+    C = [[None for i in range(col_matrix_1)] for j in range(row_matrix_2)]
```

Comment:
This itself will take O(m*n), so multi-threading is of no use.

Reply:
```suggestion
    C = [[] for j in range(row_matrix_2)]
```

Reply:
It wont help because at the end each operation is of O(1).

Reply:
> This itself will take O(m*n), so multi-threading is of no use.

and each thread will execute simultaneously so it would be O(m*n + m) for simplicity let m=n so it would be O(n^2 + n) = O(n^2) and conventional method gives us O(n^3)

Reply:
I see, won't make any difference, as each thread is filling each [i, j] so complexity won't go below  n**2 but the suggestion may reduce the constant factor.